nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -232,7 +271,11 @@ private void compareNullabilityAnnotations(
         }
       }
       if (isLHSNullableAnnotated != isRHSNullableAnnotated) {
-        reportInvalidAssignmentInstantiationError(tree, lhsType, rhsType, state, analysis);
+        if (tree instanceof AssignmentTree || tree instanceof VariableTree) {
```

Comment:
instead of putting the conditional here, how about having a single error reporting function, and checking the type of the tree in that?

Reply:
Actually, I had a better idea here.  Rather than reporting errors inside `compareNullabilityAnnotations()`, let's have `compareNullabilityAnnotations()` return a boolean which is true when the annotations are identical and false otherwise.  Then, the caller can be responsible for reporting the error, and it will be much easier to report different types of errors in different cases!

Reply:
ok. Will update this.